twitter/searcher.py

```diff
@@ -0,0 +1,182 @@
+    if not since_id:
+        since_id = get_since_id_from_file()
+    api = build_tweepy_api()
+    # tweets = []
+    total_returned_tweets = 0
+    processed_id_tweets = set()
+    max_processed_id = 0
+    max_processed_time_stamp = 0
+    for query in QUERIES:
+        cursor = query_twitter(api, query, since_id)
+        for resp in cursor:
+            total_returned_tweets += 1
+            tweet = resp._json
+            id_tweet = tweet["id"]
+            if id_tweet > max_processed_id:
+                max_processed_id = id_tweet
+                max_processed_time_stamp = tweet["created_at"]
+                # TODO: should we keep updating this file, so it is up to date if the job quits unexpectedly?
+                # pro: can pick up where we left off and avoid duplicates
+                # con: may miss hits from searches that were not yet performed
+                # log_last_processed_id(max_processed_id, max_processed_time_stamp)
+            # submissions is a list so we can handle single tweet with multiple media objects
+            submissions, processed_id_tweets = convert_tweet(tweet, processed_id_tweets)
+            if not submissions:
+                continue
+            bulk_upload_submissions(submissions, TWITTER_LARAVEL_API_KEY)
+    print("total_returned_tweets", total_returned_tweets)
+    log_last_processed_id(max_processed_id, max_processed_time_stamp)
+    return max_processed_id
+
+
+def build_tweepy_api():
+    auth = tweepy.OAuthHandler(TWITTER_CONSUMER_KEY, TWITTER_CONSUMER_SECRET)
+    auth.set_access_token(TWITTER_ACCESS_TOKEN, TWITTER_ACCESS_TOKEN_SECRET)
+    return tweepy.API(auth, wait_on_rate_limit=True)
+
+
+def query_twitter(api, query: str, since_id: int):
+    # TODO: include result_type? (mixed, recent, popular) probably want recent once batching is set up
+    # TODO: why wasn't count working
+    return tweepy.Cursor(api.search, q=query, lang="en", since_id=since_id, include_entities=True).items()
+
+
+def convert_tweet(tweet: Dict[str, Any], processed_id_tweets: Set[int]) -> Tuple[List[RawSubmission], Set[int]]:
+    submissions = []
+    # TODO: add "monetizable" logger. like, is somebody profiting off this shit?
+    if not is_tweet_relevant(tweet):
+        return None, processed_id_tweets
+    if "retweeted_status" in tweet:
+        tweet = tweet["retweeted_status"]
+    id_tweet = tweet["id"]
+    if id_tweet in processed_id_tweets:
+        return None, processed_id_tweets
+    processed_id_tweets.add(id_tweet)
+
+    converted_date = twitter_created_at_to_utc(tweet["created_at"])
+    submission_body = ""
+    if tweet["geo"]:
+        submission_body = f"Provided geo: {tweet['geo']}"
+    elif tweet["user"].get("location"):
+        submission_body = f"User Location: {tweet['user']['location']}"
+
+    for media in tweet["extended_entities"]["media"]:
+        if media["type"] != "video":
+            continue
+        print("found video for tweet: ", tweet["text"])
```

Comment:
Maybe switch to the loggers like in reddit now?

Reply:
Yea, I couldn't see the logs on kubernetes when the reddit crawler was using print statements